react/components/Shelf/index.tsx

```diff
@@ -65,10 +67,34 @@ const Shelf: StorefrontFunctionComponent<Props> = ({
     }
   }, [shelfDivRef, products, userId, correlationId, onView])
 
+  const productsIds = useMemo(
+    () => products.map((p) => p.productId).join(', '),
+    [products]
+  )
+
+  function buildProductAfDataAttributes(
+    product?: Record<string, string>,
+    index?: number
+  ) {
+    return {
+      'data-af-element': 'recommendation-shelf-product',
+      'data-af-product-id': product && product.productId,
+      'data-af-onclick': product && !!product.productId,
+      'data-af-product-position': index ?? Number(index) + 1,
+    }
+  }
+
+  const shouldAddAFAttr = !!(correlationId && campaignVrn && productsIds)
```

Comment:
ProductIds is an array so it is truthy, I think you wanted to check for productIds.length, is it not?